### PR TITLE
Enforce named constructor (argument) for Arrows3D

### LINKED ISSUE #3402: Force users to spell out the components in archetype constructors
Currently this is valid code:

```py
rr.log("base", rr.Arrows3D([0, 1, 0]))
```

As a user I would find this profoundly confusing. What are these three numbers - are they the base of the arrow, or its tip? (answer: neither!).

This would be much better if the user was forced to write out `rr.log("base", rr.Arrows3D(vector=[0, 1, 0]))`.
(In this case, I would also vote for making `origin` mandatory).

Furthermore, I think in all our examples we should include the origin for our arrows.

The same would go for Rust and C++ - `Arrows3D::from_vector(…).with_origin(…)`.

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Fixes #3402
-> as discussed, Arrows3D is the only instance where we want to change this right now. Note that we already deal with Boxes2D/Boxes3D in a similar fashion

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3411) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3411)
- [Docs preview](https://rerun.io/preview/ab405195d8852ce7bcde4e7a9d543d5aede60724/docs) 
- [Examples preview](https://rerun.io/preview/ab405195d8852ce7bcde4e7a9d543d5aede60724/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: crates/re_types/src/archetypes/boxes3d_ext.rs [hunk 1]
```diff
@@ -6,7 +6,7 @@ use crate::{
 use super::Boxes3D;
 
 impl Boxes3D {
-    /// Creates new [`Boxes3D`] with [`Self::half_sizes`] centered around the local origin.
+    /// Creates new [`Boöxes3D`] with [`Self::half_sizes`] centered around the local origin.
```
Comment (Wumpf) [inline crates/re_types/src/archetypes/boxes3d_ext.rs:9]: wait what

File: docs/code-examples/arrow3d_simple.py [hunk 2]
```diff
@@ -12,4 +12,4 @@
 vectors = np.column_stack([np.sin(angles) * lengths, np.zeros(100), np.cos(angles) * lengths])
 colors = [[1.0 - c, c, 0.5, 0.5] for c in angles / tau]
 
-rr2.log("arrows", rr2.Arrows3D(vectors, colors=colors))
+rr2.log("arrows", rr2.Arrows3D(vectors=vectors, colors=colors))
```
Comment (emilk) [inline docs/code-examples/arrow3d_simple.py:15]: Can we please add an origin here while we're at it? I think it makes the example a lot more useful to the user.

File: docs/code-examples/arrow3d_simple.rs [hunk 3]
```diff
@@ -23,7 +23,10 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         })
         .unzip();
 
-    rec.log("arrows", &Arrows3D::new(vectors).with_colors(colors))?;
+    rec.log(
+        "arrows",
+        &Arrows3D::from_vectors(vectors).with_colors(colors),
```
Comment (emilk) [inline docs/code-examples/arrow3d_simple.rs:28]: and here

File: docs/code-examples/transform3d_simple.rs [hunk 4]
```diff
@@ -12,14 +12,17 @@ use std::f32::consts::PI;
 fn main() -> Result<(), Box<dyn std::error::Error>> {
     let (rec, storage) = RecordingStreamBuilder::new("rerun_example_transform3d").memory()?;
 
-    rec.log("base", &Arrows3D::new([(0.0, 1.0, 0.0)]))?;
+    rec.log("base", &Arrows3D::from_vectors([(0.0, 1.0, 0.0)]))?;
```
Comment (emilk) [inline docs/code-examples/transform3d_simple.rs:15]: and here

File: docs/code-examples/transform3d_simple.rs [hunk 5]
```diff
@@ -12,14 +12,17 @@ use std::f32::consts::PI;
...
 
     rec.log(
         "base/translated",
         &Transform3D::new(TranslationAndMat3x3::new([1.0, 0.0, 0.0], Mat3x3::IDENTITY)),
     )?;
 
-    rec.log("base/translated", &Arrows3D::new([(0.0, 1.0, 0.0)]))?;
+    rec.log(
+        "base/translated",
+        &Arrows3D::from_vectors([(0.0, 1.0, 0.0)]),
```
Comment (emilk) [inline docs/code-examples/transform3d_simple.rs:24]: and here

File: docs/code-examples/transform3d_simple.rs [hunk 6]
```diff
@@ -30,7 +33,10 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         }),
     )?;
 
-    rec.log("base/rotated_scaled", &Arrows3D::new([(0.0, 1.0, 0.0)]))?;
+    rec.log(
+        "base/rotated_scaled",
+        &Arrows3D::from_vectors([(0.0, 1.0, 0.0)]),
```
Comment (emilk) [inline docs/code-examples/transform3d_simple.rs:38]: and here

File: examples/rust/clock/src/main.rs [hunk 7]
```diff
@@ -76,7 +76,7 @@ fn run(rec: &RecordingStream, args: &Args) -> anyhow::Result<()> {
         )?;
         rec.log(
             format!("world/{name}_hand"),
-            &Arrows3D::new([pos])
+            &Arrows3D::from_vectors([pos])
```
Comment (emilk) [inline examples/rust/clock/src/main.rs:79]: and here

File: rerun_cpp/src/rerun/archetypes/arrows3d_ext.cpp [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+// #define EDIT_EXTENSION
+
+namespace rerun {
+    namespace archetypes {
+
+#ifdef EDIT_EXTENSION
+        // [CODEGEN COPY TO HEADER START]
+
+        /// Creates new 3D arrows pointing in the given directions.
```
Comment (emilk) [inline rerun_cpp/src/rerun/archetypes/arrows3d_ext.cpp:11]:
```suggestion
        /// Creates new 3D arrows pointing in the given directions, with a base in the origin (0, 0, 0).
```

File: rerun_cpp/src/rerun/archetypes/arrows3d.hpp [hunk 9]
```diff
@@ -87,12 +87,17 @@ namespace rerun {
             static const char INDICATOR_COMPONENT_NAME[];
 
           public:
-            Arrows3D() = default;
+            // Extensions to generated type defined in 'arrows3d_ext.cpp'
 
-            Arrows3D(std::vector<rerun::components::Vector3D> _vectors)
-                : vectors(std::move(_vectors)) {}
+            /// Creates new 3D arrows pointing in the given directions.
```
Comment (emilk) [inline rerun_cpp/src/rerun/archetypes/arrows3d.hpp:92]:
```suggestion
            /// Creates new 3D arrows pointing in the given directions, with a base in the origin (0, 0, 0).
```

File: crates/re_types/src/archetypes/arrows3d_ext.rs [hunk 10]
```diff
@@ -0,0 +1,11 @@
+use crate::components::Vector3D;
+
+use super::Arrows3D;
+
+impl Arrows3D {
+    /// Creates new 3D arrows pointing in the given directions.
```
Comment (emilk) [inline crates/re_types/src/archetypes/arrows3d_ext.rs:6]:
```suggestion
    /// Creates new 3D arrows pointing in the given directions, with a base in the origin (0, 0, 0).
```

**Review by emilk — APPROVED** [review]
Thank you :pray

**Pull request merged**